### PR TITLE
ExtractorBranch

### PR DESCRIPTION
**Pull request by Ahmed-cmd-web**
### To test this script:
go to TableDataExtractor.py
then try running the script then try changing the pdf argument to pdf2

### CODE REVIEW

**Comment by OmarElsebaey001** [comment]
Kindly add the requirements.txt file that contains the required libs

**Comment by OmarElsebaey001** [comment]
Also , there should be a single function that returns the final df , in the test you provided , the test is doing some extra steps of cleaning , these steps should be included in the class

**Comment by OmarElsebaey001** [comment]
the column trading & ucc is to be divided into two

File: Task3_Extractor_Script/DataClass.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+import pdfplumber
+import pandas as pd
+from typing import Callable
+
+
+class Data:
+    def __init__(self,pdf:str,password:str='',dataFrameColumns:list=[]) -> None:
```
Comment (OmarElsebaey001) [inline Task3_Extractor_Script/DataClass.py]: kindly remove the password part

File: Task3_Extractor_Script/TableDataClass.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    def __isData(self,arr:list)->bool:
+        if len(arr)<4 or not arr[2]:
+            return False
+        res=False
+        try:
+            datetime.strptime(arr[1],"%H:%M:%S")
+            res=True
+        except :
+            return False
+        return arr[0].isdigit() and len(arr[0])>=8 and res
```
Comment (OmarElsebaey001) [inline Task3_Extractor_Script/TableDataClass.py]: what if the arr[0] is less than 8 ?
this condition should be more generic
we can check if data with a regex on the whole row

**Review by OmarElsebaey001 — COMMENTED** [review]
Bin directory is to be removed

File: Task3_Extractor_Script/TableDataClass.py (hunk 2 above)
Comment (OmarElsebaey001) [inline Task3_Extractor_Script/TableDataClass.py]: The condition of array length is still there

**Comment by OmarElsebaey001** [comment]
I am not able to find TableDataExtractor.py

File: Task3_Extractor_Script/DataClass.py (hunk 1 above)
Comment (Ahmed-cmd-web) [reply Task3_Extractor_Script/DataClass.py]: ok,but if u wish to test it you will have to unlock all the pdfs before moving forward

**Comment by Ahmed-cmd-web** [comment]
i changed it from TableDataExtractor.py to TableDataClass.py

File: Task3_Extractor_Script/DataClass.py (hunk 1 above)
Comment (OmarElsebaey001) [reply Task3_Extractor_Script/DataClass.py]: yes, have you looked in how to unlock all the pdfs ?

File: Task3_Extractor_Script/TableDataClass.py (hunk 2 above)
Comment (OmarElsebaey001) [reply Task3_Extractor_Script/TableDataClass.py]: @Ahmed-cmd-web have you looked into this?

**Pull request merged**